tests/unit/test_helper_types.py

```diff
@@ -77,23 +78,72 @@ def test_init_valid_silence_event(self) -> None:
         assert my_options.silence == all_events
 
     @pytest.mark.parametrize(
-        "include,silence,expected_includes",
+        "include,exclude,silence,expected_includes",
         [
-            (["ItemA"], ["ItemA"], False),
-            ("*", ["ItemA"], False),
-            ("*", ["ItemB"], True),
+            ([], [], [], False),
+            (["BehaviorChangeEvent"], [], ["BehaviorChangeEvent"], False),
+            (["BehaviorChangeEvent"], [], [], True),
+            ("*", ["BehaviorChangeEvent"], ["BehaviorChangeEvent"], False),
+            ("*", [], ["BehaviorChangeEvent"], False),
+            ("*", ["BehaviorChangeEvent"], [], False),
+            ("*", [], [], True),
+            ("*", ["ItemB"], [], True),
+            ("*", [], ["ItemB"], True),
+            (["BehaviorChangeEvent"], [], ["Deprecations"], True),
```

Comment:
Shouldn't this be false?  BehaviorChangeEvent is a deprecation, but Deprecations are silenced so I think it wouldn't be included?

Reply:
It should be true. Specific naming takes precedence over more general groups. That is WarnErrorOptions configuration would be "I want to silence all deprecations, except `BehaviorChangeEvent` which I want to error"

Reply:
In the docstring for the `WarnErrorOptions` class I put the following information
> In a hierarchy of configuration, the following rules apply:
>    - named > deprecations > "all"/"*"
>    - silence > exclude > include

I guess what is missing here is the former takes precedence over the later 🤔 Would adding to something to that effect make it clearer?